app/routes/edit-course.js

```diff
@@ -0,0 +1,238 @@
+const ApplicationHelper = require('../data/helpers/application')
+const CourseHelper = require('../data/helpers/courses')
+
+module.exports = router => {
+  router.get('/applications/:applicationId/course/edit/course', (req, res) => {
+    let selectedCourse
+    if (req.session.data['edit-course'] && req.session.data['edit-course'].course) {
+      selectedCourse = req.session.data['edit-course'].course
+    }
+
+    let back = `/applications/${req.params.applicationId}`
+
+    if (req.query.referrer === 'check') {
+      back += '/course/edit/check'
+    }
+
+    res.render('applications/course/course', {
+      application: req.session.data.applications.find(app => app.id === req.params.applicationId),
+      courses: CourseHelper.getCourses(selectedCourse),
+      actions: {
+        back: back,
+        cancel: `/applications/${req.params.applicationId}/course/edit/cancel`,
```

Comment:
Why is cancel a separate url?

Reply:
For convenience to clear out the session data.